# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: gfx/thebes/gfxTextRun.cpp
# Commit: 12fc56d7782c
# Full Hash: 12fc56d7782c1aed1b2975d14a3c9d9f4c04b3a3
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2020-04-08 21:42:38
# Description:
#   Bug 1627605 - Avoid creating strings with shared-memory buffers that could become unmapped on a font-list refresh. r=jwatt
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D70025
# ==============================================================================

diff -r b40b0c09c1fe -r 12fc56d7782c gfx/thebes/gfxTextRun.cpp
--- a/gfx/thebes/gfxTextRun.cpp	Wed Apr 08 09:46:34 2020 +0000
+++ b/gfx/thebes/gfxTextRun.cpp	Wed Apr 08 09:53:35 2020 +0000
@@ -1736,7 +1736,8 @@
 
   if (mFontMatchingStats) {
     for (const auto& f : fonts) {
-      nsAutoCString key;
+      nsCString key;  // not nsAutoCString, as the assignment to it won't copy
+                      // characters, it'll just share the string buffer
       FontVisibility visibility;
       if (f.mFamily.mIsShared) {
         key = f.mFamily.mShared->Key().AsString(pfl->SharedFontList());
